src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToTestAMultitargetedSolutionWithPublishReleaseOrPackRelease.cs

```diff
@@ -0,0 +1,395 @@
+               {
+                   new Tuple<string, string>(firstProjectTfm, expectedConfiguration),
+                   new Tuple<string, string>(secondProjectTfm, expectedConfiguration),
+               });
+
+            VerifyCorrectConfiguration(finalPropertyResults, expectedConfiguration);
+        }
+
+        [InlineData("net7.0", true)]
+        [InlineData("-p:TargetFramework=net7.0", false)]
+        [Theory]
+        public void ItPublishesDebugWithATargetFrameworkOptionNet7ForNet8Net7ProjectAndNet7Net6ProjectSolutionWithPublishReleaseUndefined(string args, bool passDashF)
+        {
+            var expectedTfm = "net7.0";
+            var expectedConfiguration = Debug;
+
+            var solutionAndProjects = Setup(Log, new List<string> { "net6.0", "net7.0" }, new List<string> { "net7.0", "net8.0" }, PublishRelease, "", "");
+            var sln = solutionAndProjects.Item1;
+
+            var dotnetCommand = new DotnetCommand(Log, publish);
+            dotnetCommand
+                .Execute(passDashF ? "-f" : "", args, sln.SolutionPath)
+                .Should()
+                .Pass();
+
+            var finalPropertyResults = sln.ProjectProperties(new List<Tuple<string, string>>()
+               {
+                   new Tuple<string, string>(expectedTfm, expectedConfiguration),
+                   new Tuple<string, string>(expectedTfm, expectedConfiguration),
+               });
+
+            VerifyCorrectConfiguration(finalPropertyResults, expectedConfiguration);
+        }
+
+        [Fact]
+        public void ItPublishesReleaseIfNet7DefinesPublishReleaseTrueNet8PlusDefinesNothing()
+        {
+            var firstProjectTfm = "net7.0";
+            var secondProjectTfm = ToolsetInfo.CurrentTargetFramework;
+            var expectedConfiguration = Release;
+
+            var solutionAndProjects = Setup(Log, new List<string> { firstProjectTfm }, new List<string> { secondProjectTfm }, PublishRelease, "true", "");
+            var sln = solutionAndProjects.Item1;
+
+            var dotnetCommand = new DotnetCommand(Log, publish);
+            dotnetCommand
+                .Execute(sln.SolutionPath)
+                .Should()
+                .Pass();
+
+            var finalPropertyResults = sln.ProjectProperties(new List<Tuple<string, string>>()
+               {
+                   new Tuple<string, string>(firstProjectTfm, expectedConfiguration),
+                   new Tuple<string, string>(secondProjectTfm, expectedConfiguration),
+               });
+
+            VerifyCorrectConfiguration(finalPropertyResults, expectedConfiguration);
+        }
+
+
+        [InlineData("true", PublishRelease)]
+        [InlineData("false", PublishRelease)]
+        [InlineData("", PublishRelease)]
+        [InlineData("true", PackRelease)]
+        // [InlineData("false", PackRelease)] This case we would expect to fail as PackRelease is enabled regardless of TFM.
```

Comment:
If we expect this to fail, do we have a test case covering that?  Should we?

Reply:
Yes, since `PackRelease` is enabled by default for all TFMs. I have changed the test to reflect the following.